File: dotnet/ClientFiltering/Models/GroupCriteria.cs
namespace ClientFiltering.Models;

[DataContract]
public record GroupCriteria
{
    [DataMember]
    public required string FieldName { get; init; }

    [DataMember]
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public required OrderDirections Direction { get; init; }

    [DataMember]
    public IReadOnlyCollection<AggregateCriteria>? Aggregates { get; init; }

    [DataMember]
    public GroupCriteria? SubGroup { get; init; }
}

File: javascript/src/lib/groupcriteria.ts
import { OrderDirections } from "./enums";
import { AggregateCriteria } from "./aggregatecriteria";

export class GroupCriteria {
  public FieldName: string;
  public Direction = OrderDirections.Ascending;
  public Aggregates: AggregateCriteria[] = [];

  constructor(fieldName = "") {
    this.FieldName = fieldName;
  }
}
